code/mission/missionparse.cpp

```diff
@@ -7100,13 +7100,13 @@ void mission_eval_arrivals()
 				if (rship >= 0)
 				{
 					int j;
-					char message_name[NAME_LENGTH + 10];
+					SCP_string message_name;
 					sprintf(message_name, "%s Arrived", wingp->name);
```

Comment:
Was this file intended to be part of this PR?  And incidentally, does sprintf support SCP_string?

Reply:
This is intentional. GCC 9 detects more possible buffer overruns when using `sprintf` so I also needed to fix that. `sprintf` support for `SCP_string` is part of FSO and was implemented some time ago.